### ISSUE TITLE
Failed to open -: unknown file type

### ISSUE THREAD

**Issue by christacaggiano** [heathsc/gemBS#42]
Hi, 

When I am running gemBS call I get the following error `Failed to open -: unknown file type` 

```bash 
: ----------- Methylation Calling --------
: Reference       : reference/sacCer3.fa.gz
: Species         : Yeast
: Right Trim      : 0
: Left Trim       : 5
: Chromosomes     : ['@pool_1', '@pool_2', '@pool_3']
: Threads         : 1
: Sample: A001XS1    Bam: ./mapping/A001XS1/A001XS1.bam
: Sample: A001XS2    Bam: ./mapping/A001XS2/A001XS2.bam
: Sample: A001XS3    Bam: ./mapping/A001XS3/A001XS3.bam
: Sample: A001XS4    Bam: ./mapping/A001XS4/A001XS4.bam
: Sample: A001XS5    Bam: ./mapping/A001XS5/A001XS5.bam
:
: Methylation Calling...
2018-12-17 18:20:53,695 ERROR: Process '/ye/zaitlenlabstore/christacaggiano/miniconda3/envs/gembs/lib/python3.6/site-packages/gemBS/bin/bcftools' finished with 255
2018-12-17 18:20:53,696 ERROR: Failed to open -: unknown file type
Exception in thread Thread-1:
Traceback (most recent call last):
ValueError: Error while executing the bscall process.
``` 

this seems to be specifically a bscall issue, but I can't seem to find what would be causing this issue. The same error comes up with the sample files and my own data. Do you have any insight on what may be causing this error?

**Comment by karl616** [comment]
I get the same error running the singularity container. For me it seems tied to the machine I'm using. When it fails it generates this output:

[stderr.txt](https://github.com/heathsc/gemBS/files/2983087/stderr.txt)
[stdout.txt](https://github.com/heathsc/gemBS/files/2983088/stdout.txt)

I should say that this is not the test data, but I'm perplexed as it works on one system.

**Comment by heathsc** [comment]
Can you send me the output you get from running:

gemBS --loglevel debug call

Thanks

**Comment by karl616** [comment]
Of course:

[debug.txt](https://github.com/heathsc/gemBS/files/2983281/debug.txt)

I can see the command that fails. I will try it step by step...

**Comment by heathsc** [comment]
You seem to be triggering a bug in bs_call.  If you could find a BAM (or a short fragment of a BAM) that triggers the problem and share it with me then I will be able to track this down.

Simon

**Comment by karl616** [comment]
I will do that, but it will take a little longer...  we deleted the alignments here by mistake... I will regenerate them...

Another thought, can something happen when loading the reference? When just running the job up until bs_call, I had the feeling it didn't get past this step.

**Comment by heathsc** [comment]
Aside from running out of memory while reading in the reference I can’t see too many places where it could fail.

Simon

**Comment by karl616** [comment]
OK, here are the first 1000 rows from a file that fails:

[test.sam.txt](https://github.com/heathsc/gemBS/files/2983826/test.sam.txt)

Inside the container this command fails:

```
cat test.sam.txt | /usr/local/lib/python3.5/dist-packages/gemBS/gemBSbinaries/bs_call -r /vol/data/reference/mm10/wgbs/mm10_no_alt_analysis_set_ENCODE.fasta.gz -n liver_late_2 --contig-bed /vol/data/test_WGBS/output/calls/liver_late_2/contigs_liver_late_2_chr18.bed --report-file /vol/data/test_WGBS/output/calls/liver_late_2/liver_late_2_chr18.json --right-trim 0 --left-trim 5 --conversion 0.0100,0.0500 --reference-bias 2 --mapq-threshold 10 --bq-threshold 13
```
but not on all machines.

**Comment by Smart-zhi** [comment]
Hello, I had the same problem. Has this problem been solved? If it possible please tell me how to solve it.

**Comment by heathsc** [comment]
Good morning,

Can you confirm the version of gemBS that you are using, and at what stage of the process  you see the problem?

Thanks,
Simon

**Comment by Smart-zhi** [comment]
Hello Simon,
  My apologies for the delay in responding.  I'm using `gemBS 3.3.0`.
  When I ran my data, I got ` AttributeError: 'list' object has no attribute 'lower' ` by using `gemBS call`.  Then, I used `gemBS call -C 0.01,0.05 `,and I got this issue：

```
: 
: Command call started at 2019-07-02 22:01:30.985422
: 
: ----------- Methylation Calling --------
: Reference       : reference/Arabidopsis_thaliana.fa
: Species         : Arabidopsis
: Right Trim      : 0
: Left Trim       : 5
: Chromosomes     : ['NC_003070.9', 'NC_003076.8', '@pool_1', '@pool_2', '@pool_3']
: Threads         : 12
: Sample: A1    Bam: ./mapping/A1.bam
: 
: Methylation Calling...
2019-07-02 22:01:31,144 ERROR: Process '/home/zhangzhi/.local/lib/python3.7/site-packages/gemBS/bin/bcftools' finished with 255
2019-07-02 22:01:31,144 ERROR: Failed to open -: unknown file type
2019-07-02 22:01:31,144 ERROR: guments)
2019-07-02 22:01:31,144 ERROR: c (conversion) option requires two comma separated arguments
Exception in thread Thread-1:
Traceback (most recent call last):
ValueError: Error while executing the bscall process.

2019-07-02 22:01:31,151 ERROR: Process '/home/zhangzhi/.local/lib/python3.7/site-packages/gemBS/bin/bcftools' finished with 255
2019-07-02 22:01:31,151 ERROR: Failed to open -: unknown file type
2019-07-02 22:01:31,151 ERROR: guments)
2019-07-02 22:01:31,151 ERROR: c (conversion) option requires two comma separated arguments
Exception in thread Thread-2:
Traceback (most recent call last):
ValueError: Error while executing the bscall process.

2019-07-02 22:01:31,411 ERROR: Process '/home/zhangzhi/.local/lib/python3.7/site-packages/gemBS/bin/bcftools' finished with 255
2019-07-02 22:01:31,412 ERROR: Failed to open -: unknown file type
Exception in thread Thread-3:
Traceback (most recent call last):
ValueError: Error while executing the bscall process.

2019-07-02 22:01:31,419 ERROR: Process '/home/zhangzhi/.local/lib/python3.7/site-packages/gemBS/bin/bcftools' finished with 255
2019-07-02 22:01:31,419 ERROR: Failed to open -: unknown file type
2019-07-02 22:01:31,419 ERROR: guments)
2019-07-02 22:01:31,420 ERROR: c (conversion) option requires two comma separated arguments
Exception in thread Thread-4:
Traceback (most recent call last):
ValueError: Error while executing the bscall process.

: Methylation call done, samples performed: A1
```

Thanks,
Zhi

**Comment by Smart-zhi** [comment]
Hello Simon, I have solved my problem. I modified `gemBS.json` in `./.gemBS/ `.
``` json
{
  "config": {

    "calling": {
      "conversion": "0.01,0.05",
    },

  }
}
```

**Comment by heathsc** [comment]
Thank you - this is a work around (because if you specify the conversion in the config file, you don’t need to set it in on the command line), but I will fix the original problem as well.

Thanks for the report.
Simon

**Comment by heathsc** [comment]
Fixed in 3.3
